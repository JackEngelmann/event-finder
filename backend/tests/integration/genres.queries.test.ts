import {
    genreFragment,
    ApolloTestServer,
    createApolloTestServer,
} from '../utils'

let apolloTestServer: ApolloTestServer | undefined

const DB_NAME = 'genresquerydb'

beforeEach(async () => {
    apolloTestServer = await createApolloTestServer({
        isAdmin: false,
        dbName: DB_NAME,
        insertTestData: true,
    })
})

afterEach(async () => {
    if (apolloTestServer) await apolloTestServer.destroy()
    apolloTestServer = undefined
})

describe('club queries', () => {
    test('genres correctly', async () => {
        const result = await apolloTestServer!.client.query({
            query: `
            {
                genres { ${genreFragment} }
            }
            `,
        })
        expect(result.data).toBeDefined()
        expect(result).toMatchSnapshot()
    })
})
