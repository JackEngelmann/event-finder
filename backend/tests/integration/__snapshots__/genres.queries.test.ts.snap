// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`club queries genres correctly 1`] = `
Object {
  "data": Object {
    "genres": Array [
      Object {
        "id": 1,
        "name": "genre1",
      },
      Object {
        "id": 2,
        "name": "genre2",
      },
      Object {
        "id": 3,
        "name": "genre3",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
